fastly/validators.go

```diff
@@ -179,3 +180,14 @@ func validatePEMBlocks(pemType string) schema.SchemaValidateDiagFunc {
 		return nil, nil
 	})
 }
+
+func validateStringTrimmed() schema.SchemaValidateDiagFunc {
```

Comment:
When I last had a crack at implementing this, I used the following function:
```go
// validateStringTrimmed is a SchemaValidateDiagFunc which checks whether the string is trimmed.
func validateStringTrimmed(value interface{}, p cty.Path) diag.Diagnostics {
    if value != strings.TrimSpace(value.(string)) {
        return diag.FromErr(p.NewErrorf("string must not contain trailing whitespace"))
    }

    return nil
}
```

It works either way, but for a brand new validation function, it could make sense to use the `schema.SchemaValidateDiagFunc` type directly rather than using the legacy `ToDiagFunc` conversion function like the others do. Other than that, your error message is much better than the one I had!

Reply:
Thanks @bengesoff
I've modified the function a bit.